pilot/pkg/features/pilot.go

```diff
@@ -261,13 +261,13 @@ var (
 		"If enabled, Pilot will keep track of old versions of distributed config for this duration.",
 	).Get()
 
-	enableEndpointSliceController, endpointSliceControllerSpecified = env.Register(
+	EnableEndpointSliceController = env.Register(
 		"PILOT_USE_ENDPOINT_SLICE",
-		false,
+		true,
```

Comment:
Why change this? we can just use Endpoints on older than v1.21 and EndpointSlice on 1.21+.

Then this change just removes the ability to use Endpointslice on v1.20 which is not a big deal. As is, this PR breaks all users on 1.20, and we are aiming to support it for at least another release.